spec/integration/time_based_rotation_with_stale_write_spec.rb

```diff
@@ -26,7 +26,7 @@
     clean_remote_files(prefix)
     subject.register
     subject.multi_receive_encoded(batch)
-    sleep(1) # the periodic check should have kick int
+    sleep(5) # the periodic check should have kick in
```

Comment:
give it a bit more time to do his stuff before doing any assert seems to reduce a lot the noise.